src/json.ts

```diff
@@ -103,9 +148,9 @@ export function isJsonRpcNotification<T>(
  *
  * @param requestOrNotification - The JSON-RPC request or notification to check.
  */
-export function assertIsJsonRpcNotification<T>(
-  requestOrNotification: JsonRpcNotification<T> | JsonRpcRequest<T>,
-): asserts requestOrNotification is JsonRpcNotification<T> {
+export function assertIsJsonRpcNotification(
+  requestOrNotification: unknown,
+): asserts requestOrNotification is JsonRpcNotification<JsonRpcParams> {
   if (!isJsonRpcNotification(requestOrNotification)) {
     throw new Error('Not a JSON-RPC notification.');
```

Comment:
What do we feel about including more detail as to why this isn't a notification object? If not now do we plan on updating this in the future?

Reply:
Good point! Will look into better error handling.

Reply:
For now I just added `error.message` to the errors we throw. This results in errors like this:

```
Not a JSON-RPC request: At path: id -- Expected the value to satisfy a union of `number | string`, but received: undefined.
```

We could maybe improve it with custom error messages, but I think this will do for the time being.